### PR TITLE
feat: add conversion from u256 to rational

### PR DESCRIPTION
**Pull request by dmoka**
Add conversion from u256 to rational to be used in hydradx node omnipool poracle price adapter

### CODE REVIEW

File: src/ema/math.rs [hunk 1]
```diff
@@ -236,25 +236,37 @@ pub(super) fn round((n, d): (U512, U512), rounding: Rounding) -> (U512, U512) {
...
-    };
-    EmaPrice::new(n, d)
+    round_uint_to_rational!(n, d, rounding)
+}
+
+/// Round a 256 bit rational number to a 128 bit rational number.
+/// The rounding is done by shifting which implicitly rounds down both numerator and denominator.
+/// This can effectivly round the complete rational number up or down pseudo-randomly.
+/// Specify `rounding` other than `Nearest` to round the whole number up or down.
+pub fn round_u256_to_rational((n, d): (U256, U256), rounding: Rounding) -> EmaPrice {
```
Comment (apopiak) [inline src/ema/math.rs:268]: this is a general function, maybe we should move it to the general math tools?

**Review by apopiak — APPROVED** [review]

File: src/ema/math.rs (hunk 1 above)
Comment (enthusiastmartin) [reply src/ema/math.rs:268]: there should already one, or something similar in src/support/rational.

**Review by enthusiastmartin — CHANGES_REQUESTED** [review]
There is the same in src/support/rational.

as alex suggested, this shouldnt be in ema but in rather generic support math. 

suggesting to update the one in src/support/rational if needed.

File: src/ema/math.rs (hunk 1 above)
Comment (dmoka) [reply src/ema/math.rs:268]: As the rational conversion from src/support/rational is perfectly fine for my case, I will close this PR.

**Comment by dmoka** [comment]
Closed as the already existing logic from support package is good enough.

**Pull request closed without merging**